src/ui/ResultLayout/ResultLayout.ts

```diff
@@ -0,0 +1,158 @@
+import {Component} from '../Base/Component';
+import {ComponentOptions} from '../Base/ComponentOptions';
+import {IComponentBindings} from '../Base/ComponentBindings';
+import {QueryEvents} from '../../events/QueryEvents';
+import {Initialization} from '../Base/Initialization';
+import {Assert} from '../../misc/Assert';
+import {ResultListEvents, IChangeLayoutEventArgs} from '../../events/ResultListEvents';
+import {ResultLayoutEvents, IResultLayoutPopulateArgs} from '../../events/ResultLayoutEvents';
+import {$$} from '../../utils/Dom';
+import {IQueryErrorEventArgs, IQuerySuccessEventArgs} from '../../events/QueryEvents';
+import {QueryStateModel, QUERY_STATE_ATTRIBUTES} from '../../models/QueryStateModel';
+import {MODEL_EVENTS, IAttributesChangedEventArg} from '../../models/Model';
+
+export interface IResultLayoutOptions {
+}
+
+// Please keep ResultLayout.validLayouts in sync with this
```

Comment:
Remove that comment. Instead, use the type system to ensure that we must keep them in sync (and the compiler should fail to build if we don't)